x/wasm/types/tx.go

```diff
@@ -478,5 +478,165 @@ func (msg MsgUpdateParams) GetSignBytes() []byte {
+		panic(err.Error())
+	}
+	return []sdk.AccAddress{authority}
+}
+
+func (msg MsgPinCodes) GetSignBytes() []byte {
+	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(&msg))
+}
+
+func (msg MsgPinCodes) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(msg.Authority); err != nil {
+		return errorsmod.Wrap(err, "authority")
+	}
+	if len(msg.CodeIDs) == 0 {
+		return errorsmod.Wrap(sdkerrors.ErrInvalidRequest, "empty code ids")
+	}
+	return nil
+}
+
+func (msg MsgUnpinCodes) Route() string {
+	return RouterKey
+}
+
+func (msg MsgUnpinCodes) Type() string {
+	return "unpin-codes"
+}
+
+func (msg MsgUnpinCodes) GetSigners() []sdk.AccAddress {
+	authority, err := sdk.AccAddressFromBech32(msg.Authority)
+	if err != nil { // should never happen as valid basic rejects invalid addresses
+		panic(err.Error())
+	}
+	return []sdk.AccAddress{authority}
+}
+
+func (msg MsgUnpinCodes) GetSignBytes() []byte {
+	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(&msg))
+}
+
+func (msg MsgUnpinCodes) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(msg.Authority); err != nil {
+		return errorsmod.Wrap(err, "authority")
+	}
+	if len(msg.CodeIDs) == 0 {
+		return errorsmod.Wrap(sdkerrors.ErrInvalidRequest, "empty code ids")
+	}
+	return nil
+}
+
+func (msg MsgSudoContract) Route() string {
+	return RouterKey
+}
+
+func (msg MsgSudoContract) Type() string {
+	return "sudo-contract"
+}
+
+func (msg MsgSudoContract) GetSigners() []sdk.AccAddress {
+	authority, err := sdk.AccAddressFromBech32(msg.Authority)
+	if err != nil { // should never happen as valid basic rejects invalid addresses
+		panic(err.Error())
+	}
+	return []sdk.AccAddress{authority}
+}
+
+func (msg MsgSudoContract) GetSignBytes() []byte {
+	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(&msg))
+}
+
+func (msg MsgSudoContract) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(msg.Authority); err != nil {
+		return errorsmod.Wrap(err, "authority")
+	}
+	if _, err := sdk.AccAddressFromBech32(msg.Contract); err != nil {
+		return errorsmod.Wrap(err, "contract")
+	}
+	if err := msg.Msg.ValidateBasic(); err != nil {
+		return errorsmod.Wrap(err, "payload msg")
+	}
+	return nil
+}
+
+func (msg MsgStoreAndInstantiateContract) Route() string {
+	return RouterKey
+}
+
+func (msg MsgStoreAndInstantiateContract) Type() string {
+	return "store-and-instantiate-contract"
+}
+
+func (msg MsgStoreAndInstantiateContract) GetSigners() []sdk.AccAddress {
+	authority, err := sdk.AccAddressFromBech32(msg.Authority)
+	if err != nil { // should never happen as valid basic rejects invalid addresses
+		panic(err.Error())
+	}
+	return []sdk.AccAddress{authority}
+}
+
+func (msg MsgStoreAndInstantiateContract) GetSignBytes() []byte {
+	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(&msg))
+}
+
+func (msg MsgStoreAndInstantiateContract) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(msg.Authority); err != nil {
+		return errorsmod.Wrap(err, "authority")
+	}
+
```

Comment:
Use `ValidateVerificationInfo(p.Source, p.Builder, p.CodeHash)` as in `StoreAndInstantiateContractProposal` for optional fields